### PR TITLE
Unify help

### LINKED ISSUE #23: Find a way to fix documentation duplication
### Feature Description
Our current documentation is split into three pieces:
1. function docs
2. matlabbatch docs
3. user manual

We maintain these three pieces by literally copy-pasting the documentation. Since this is very inefficient and error-prone, a better way to handle this process is required.

### PR DESCRIPTION
**Pull request by dominikbach**
Fixes #23  .

Changes proposed in this pull request: Matlabbatch GUI items follow the logic of using function help wherever possible. 
Exceptions:
- generic help texts in `pspm_cfg_selector_...` functions
- GUI-specific hints and help
- data-type specific help in `pspm_import`, which is partly summarised in `pspm_text`. This will be dealt with in a separate PR.
- `pspm_review` which has no function help
- `pspm_process_illuminance`: function help needs to be modified for `pspm_help` 
- `pspm_pupil_correct`: function help needs to be modified for `pspm_help`

### CODE REVIEW

**Comment by teddychao** [comment]
image

I suggest that I can update `pspm_cfg_help_format` to make sure the output will not start with a blank space or a dot.

File: src/pspm_cfg/pspm_cfg_help_format.m [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            '------------------------------------------------------------------------------------------------';...
+            'References:'; ...
+            settings.help.(funcname).References(:)];
+        end
+    end
+else
+    % this syntax allows chaining several nested structs into one argname
+    evalc(sprintf('helptext = settings.help.%s.Arguments.%s;', funcname, argname));
+    [startindx, endindx] = regexp(helptext, '\[\s*([^\[\]]*)\s*\]'); % thanks ChatGPT for finding the regexp
+    helptext(startindx:endindx) = [];
```
Comment (teddychao) [inline src/pspm_cfg/pspm_cfg_help_format.m:37]: `startindx` and `endindx` could be 1xn (n>1) array. I would suggest to use `transpose` in this occasion. Please find screenshot.

image

File: src/pspm_cfg/pspm_cfg_pp_heart_data.m [hunk 2]
```diff
@@ -66,85 +66,74 @@
...
 
 ecg2hb_amri_teo_bandpass         = cfg_entry;
 ecg2hb_amri_teo_bandpass.name    = 'TEO bandpass filter';
 ecg2hb_amri_teo_bandpass.tag     = 'teo_bandpass';
 ecg2hb_amri_teo_bandpass.strtype = 'r';
 ecg2hb_amri_teo_bandpass.num     = [1 2];
 ecg2hb_amri_teo_bandpass.val     = {[8 40]};
-ecg2hb_amri_teo_bandpass.help    = {['Define the cutoff frequencies (Hz) for bandpass filter applied to filtered ECG',...
-    ' signal before applying TEO']};
+ecg2hb_amri_teo_bandpass.help    = pspm_cfg_help_format('pspm_convert_ecg2hb_amri', 'options.teo_bandpass');
```
Comment (teddychao) [inline src/pspm_cfg/pspm_cfg_pp_heart_data.m:93]: Just now sure if this is expected...

image

File: src/pspm_combine_markerchannels.m [hunk 3]
```diff
@@ -18,6 +18,7 @@
 %   │                   the first option is used, then use marker channel
 %   │                   indexing in further processing which by default
 %   │                   takes the first marker channel as input
+%   └.marker_chan_num:  any number of marker channel numbers - if undefined
 %   └.marker_chan_num : any number of marker channel numbers - if undefined
```
Comment (teddychao) [inline src/pspm_combine_markerchannels.m:22]: There are two duplicated lines?

File: src/pspm_emg_pp.m [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
-%   .channel_action:  ['add'/'replace'] Defines whether the new channel should
+%   ┌─────── options:
+%   ├────.mains_freq: [integer] Frequency of mains noise to remove
+%   │                 with notch filter (default: 50 Hz).
+%   ├───────.channel: [numeric/string] Channel to be preprocessed.
+%   │                 Can be a channel ID or a channel name.
+%   │                 Default is 'emg' (i.e. last EMG channel)
+%   └.channel_action: ['add'/'replace'] Defines whether the new channel should
 %                     be added or the previous outputs of this function should
 %                     be replaced. (Default: 'replace')
```
Comment (teddychao) [inline src/pspm_emg_pp.m:27]: There is an empty line above `History` section. Better to delete it to avoid potential mis-recognition.

**Review by teddychao — APPROVED** [review]
Just minor changes could be useful....

File: src/pspm_cfg/pspm_cfg_help_format.m (hunk 1 above)
Comment (dominikbach) [reply src/pspm_cfg/pspm_cfg_help_format.m:37]: fixed.

File: src/pspm_cfg/pspm_cfg_pp_heart_data.m (hunk 2 above)
Comment (dominikbach) [reply src/pspm_cfg/pspm_cfg_pp_heart_data.m:93]: fixed.

File: src/pspm_combine_markerchannels.m (hunk 3 above)
Comment (dominikbach) [reply src/pspm_combine_markerchannels.m:22]: fixed.

File: src/pspm_emg_pp.m (hunk 4 above)
Comment (dominikbach) [reply src/pspm_emg_pp.m:27]: fixed.

**Pull request merged**